### PR TITLE
Feature/pupil spa login

### PR DESCRIPTION
**Pull request by jon-shipley**
Adds login, login success, login failures and a UserService

### CODE REVIEW

**Comment by jon-shipley** [comment]
@GeorgeChatzigiannis - for routing.  The UserService keeps track of whether the user is logged in or now.

**Comment by GeorgeChatzigiannis** [comment]
@jon-shipley is there any possible way to pass down state(isSuccessfulLogin) to determine what should the component render?

**Comment by GeorgeChatzigiannis** [comment]
@jon-shipley I see the angular routing can accept further arguments that could potentially help determining what to render. Example: https://stackoverflow.com/questions/36835123/how-do-i-pass-data-in-angular-2-components-while-using-routing

**Comment by activemq** [comment]
JS tests added

30 scenarios (30 passed)
89 steps (89 passed)
0m46.433s

@jon-shipley @pris54 @GeorgeChatzigiannis @shekss please review.

**Comment by jon-shipley** [comment]
Ready for review - all unit tests pass

Code coverage complete
screen shot 2017-08-01 at 14 54 05

**Comment by GuyHarwood** [comment]
happy llama contest picture
Happy Llama approves

File: pupil-spa/src/app/login-success/login-success.component.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    <p class="lede">
+      <a routerLink="/sign-out" >This is not me, I will enter my details again</a>
+    </p>
+    <form action="/check/start" method="GET">
+      <div class="form-group">
+        <input type="submit" class="button" value="Read Instructions">
+      </div>
+    </form>
+  </div>
+</div>
```
Comment (pris54) [inline pupil-spa/src/app/login-success/login-success.component.html]: New line?

File: pupil-spa/src/app/login.response.mock.json [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  },
+  "school": {
+    "id": 9991001,
+    "name": "Example School One"
+  },
+  "config": {
+    "questionTime": 5,
+    "loadingTime": 2
+  }
+}
```
Comment (pris54) [inline pupil-spa/src/app/login.response.mock.json]: New line missing?

File: test/data/config.json [hunk 3]
```diff
@@ -1,6 +1,7 @@
 {
   "test": {
     "adminUrl": "http://localhost:3001/",
-    "checkUrl": "http://localhost:3000/"
+    "checkUrl": "http://localhost:3000/",
+    "spaUrl": "http://localhost:4200/"
   }
 }
```
Comment (pris54) [inline test/data/config.json]: New line?

File: pupil-spa/src/app/login/login.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+      navigate: jasmine.createSpy('navigate')
+    };
+
+    mockUserService = {
+      login: function() { return undefined; }
+    };
+
+    const loginPromise = new Promise((resolve, reject) => {
+      promiseHelper = {
+        resolve: resolve,
```
Comment (GeorgeChatzigiannis) [inline pupil-spa/src/app/login/login.component.spec.ts]: property shorthand here, can be:
resolve,
reject

**Comment by jon-shipley** [comment]
Review comments addressed.

**Pull request merged**